### PR TITLE
store/tikv: Support RawReverseScan (#8352)

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve? 

This PR cherry-picks #8352 to release-2.1. This PR provides reversed query operations of RegionCache. However this PR intends to implement RawReverseScan, so for completeness concern, I recommend that merge https://github.com/pingcap/kvproto/pull/434 
 and https://github.com/tikv/tikv/pull/5148 first.

### What is changed and how it works?

This PR cherry-picks #8352 .

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - No

### CODE REVIEW

**Comment by winkyao** [comment]
What is the corresponding parser pr?

**Comment by MyonKeminta** [comment]
@winkyao This PR seems to be not related to parser..?

**Comment by MyonKeminta** [comment]
/run-all-tests tikv=pr/5149 pd=release-2.1 tidb-test=release-2.1

**Comment by MyonKeminta** [comment]
/run-all-tests tikv=pr/5148 pd=release-2.1 tidb-test=release-2.1

**Comment by MyonKeminta** [comment]
/run-all-tests tikv=pr/5148 pd=release-2.1 tidb-test=release-2.1

**Comment by MyonKeminta** [comment]
/run-unit-test tikv=pr/5148 pd=release-2.1 tidb-test=release-2.1

**Comment by MyonKeminta** [comment]
/run-all-tests tikv=release-2.1 pd=release-2.1 tidb-test=release-2.1

**Comment by MyonKeminta** [comment]
/run-all-tests tikv=release-2.1 pd=release-2.1 tidb-test=release-2.1
@winkyao @crazycs520 PTAL

File: store/tikv/rawkv.go [hunk 1]
```diff
@@ -306,11 +306,63 @@ func (c *RawKVClient) Scan(startKey []byte, limit int) (keys [][]byte, values []
...
+
+	if limit > MaxRawKVScanLimit {
+		return nil, nil, errors.Trace(ErrMaxScanLimitExceeded)
+	}
+
+	for len(keys) < limit {
+		req := &tikvrpc.Request{
+			Type: tikvrpc.CmdRawScan,
+			RawScan: &kvrpcpb.RawScanRequest{
+				StartKey: startKey,
```
Comment (zz-jason) [inline store/tikv/rawkv.go]: forget to set `Endkey`?

Comment (MyonKeminta) [reply store/tikv/rawkv.go]: The original PR is merged after another PR that supports EndKey, which is not cherry-picked to release-2.1. Here the EndKey was removed manually.

Comment (MyonKeminta) [reply store/tikv/rawkv.go]: Tests about EndKey is also removed.

Comment (zz-jason) [reply store/tikv/rawkv.go]: should we cherry pick that pr to release 2.1 as well?

Comment (MyonKeminta) [reply store/tikv/rawkv.go]: I don't think so. There are no requirement of that feature in release 2.1, but cherry-picking it needs cherry-picking another PR in kvproto and another PR in TiKV. Raw reverse scan is not required too. The reason of cherry-picking this PR is that reverse querying support of RegionCache is needed.

Comment (zz-jason) [reply store/tikv/rawkv.go]: okay

**Pull request merged**